File: evals/seeded-py/shop/pricing.py
def line_total(item):
    """Total for one line item.

    item: {"unit_price": number, "qty": number,
           "discount": optional fraction 0-1}
    """
    price = float(item["unit_price"])
    qty = item["qty"]
    discount = item.get("discount", 0)
    return round(price * qty * (1 - discount), 2)

File: evals/seeded-ts/src/pricing.ts
export interface LineItem {
  unitPrice: number;
  qty: number;
  discount?: number;
}

export function lineTotal(item: LineItem): number {
  const discount = item.discount ?? 0;
  return Math.round(item.unitPrice * item.qty * (1 - discount) * 100) / 100;
}

export function parseOrder(raw: string): LineItem[] {
  return JSON.parse(raw) as LineItem[];
}
